unitTests/CMakeLists.txt

```diff
@@ -1,4 +1,5 @@
-find_package(GTest REQUIRED)
+# Prefer using Google suplied CMake config as bult-in module is buggy before 3.23
+find_package(GTest REQUIRED CONFIG)
```

Comment:
If gtest supplies COMPONENTS, then declare a components required. 

In the case that gtest is installed but gmock is not installed, the find_package returns found but the gmock namespace target wont exist. 

`find_package(GTest REQUIRED COMPONENTS gtest gmock gtest_main gmock_main)`

There's some churn around the target names depending on how it was found, but that's the idea.

Reply:
I had a qucik look already beforehand, I don't think it does...

Reply:
But there's a different problem: at least on Fedora, both `gmock` (pulled in by `gtest-devel`) and `gmock-devel` ship the shared objects (the `gmock-devel` just the unversioned symlink though), so this actually doesn't catch the missing `gmock-devel`...

Reply:
FWIW for meson, I use gmock_main as the dependency.

Reply:
I tried using components on my branch `use-gtest-components` like so:
`find_package(GTest CONFIG COMPONENTS gtest_main gmock_main)`

On Ubuntu 22.04 with `libgmock-dev` istalled, the package found returns successfully.

If I add a bogus component `foobar`, the gtest find package call reports the component as found. Seems like a bug to me in their find module.